src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/StorageSection/CustomizeSource/SelectSource.tsx

```diff
@@ -72,87 +74,110 @@ export const SelectSource: FC<SelectSourceProps> = ({
     control,
     formState: { errors },
     register,
-    watch,
   } = useFormContext();
-  const httpURL = watch(`${testId}-httpURL`);
-  const containerImage = watch(`${testId}-containerImage`);
 
-  const [volumeQuantity, setVolumeQuantity] = useState(initialVolumeQuantity);
+  const initialDiskSource = useRef(selectedSource);
+
+  const volumeQuantity = (selectedSource as V1beta1DataVolumeSpec)?.storage?.resources?.requests
```

Comment:
here i deleted all the useStates. The state is represented from only the template/vm in the context and all the data comes from there.